### PR TITLE
feat(#1029): Add repository checklist

### LINKED ISSUE #1029: Document new CHT Product repository creation guidelines and repository checklist
1. Create guidelines for new GitHub repositories under Medic organization:
* **Who** can create a repository?
* Is it OK to create a CHT/Medic-related work repository under a **personal GitHub account**?
* When to make a repo **public** vs **private** vs directory in existing (**monolithic**)?

2. Put together a checklist of features a CHT-related repo should have:
* Basic CI build
* Branch protection
* License
* Issue template
* Pre-commit hook
* Secrets check 
* `README` must-haves

3. Create a `cht-template` [GitHub template repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) with the configurations above.

### PR DESCRIPTION
**Pull request by andrablaj**
Checklist to consider when creating CHT Product repositories under Medic's GitHub account.

Closes #1029

### CODE REVIEW

**Comment by andrablaj** [comment]
@mrjones-plip as we discussed yesterday, I started putting together the new repository checklist. Please review and update it as you see fit.

I created a placeholder for the FAQ questions you wanted to clarify so you can add some suggestions to this same PR. We can then send it for review and discussion with other colleagues.

**Comment by andrablaj** [comment]
@garethbowen, with @mrjones-plip, we considered having guidelines around GitHub repositories that live under the Medic organization. The questions we were hoping to find answers to are listed in #1029 (and marked as `TODOs` in this draft PR).

Could you support us with a first draft-PR review for the repository checklist ideas and support us in getting the answers to the `TODO` questions? Also, feel free to commit directly to this branch if it makes it simpler. Thank you!

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:3]: Maybe "Repository Creation Checklist" is better?

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:25]: We've discussed internally and haven't agreed on coverage targets. This is probably a separate conversation. To keep this change uncontroversial, let's drop it.

```suggestion
- [ ] Unit tests cover the majority of the code.
```

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:30]:
```suggestion
- [ ] `main` branch is always shippable.
```

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:31]: Not sure about this one. So far the plans are for CD for commits to `main` which pushes to the production market. Why use CD before merge?

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:35]:
```suggestion
- [ ] A reviewer for a PR merge is enforced by policy.
```

**Review by garethbowen — CHANGES_REQUESTED** [review]

File: content/en/contribute/code/repository-checklist.md
Comment (mrjones-plip) [reply content/en/contribute/code/repository-checklist.md:3]: This is a great point, thanks @garethbowen! We should consider a rewrite.

However, I think I have a more nascent concern for both product and repositories: We should always strive to help a real customer/solve a real problem as early as possible.  For example, if [CHT Monitoring](https://github.com/medic/cht-monitoring), which I would consider a product, needed to meet all the "at least do the following" items, we wouldn't be live now and it may be weeks or months until we met all the requirements on that list.  I don't think the investment in checking all those boxes would be worth if it because A) the lost time on helping actual customers and B) we may have a failed product in 3 months and the month we spent doing the checklist was a waste.

The list is actually great though!  I think we can do the rewrite Gareth suggested and greatly reduce the "must haves" and introduce a  "nice to have" list maybe?  The must have list might be:

* The main branch is locked via [branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule).
* Merges are done through PRs.
* PRs reference related issues.
* Commit formats follow the [guidelines](http://localhost:1313/contribute/code/workflow/#commits).
* Secrets are not part of the commit history or made public.
* For public repositories the main branch should contain a `license` and `readme.md`
*  `main` branch is always shippable.
*  A minimum number of reviewers (usually 2) for a PR merge is enforced by policy

File: content/en/contribute/code/repository-checklist.md
Comment (mrjones-plip) [inline content/en/contribute/code/repository-checklist.md:20]: A bit silly to have a license for a private repo, but it's really nice to have both, always:

```suggestion
- [ ] For repositories the `main` branch should contain the following files:
```

File: content/en/contribute/code/repository-checklist.md
Comment (mrjones-plip) [reply content/en/contribute/code/repository-checklist.md:45]: Seconded! Low barrier to entry solves problems now.  Let's rescind access when we have a problem later.

File: content/en/contribute/code/repository-checklist.md
Comment (mrjones-plip) [reply content/en/contribute/code/repository-checklist.md:51]: Agreed!  Also, if they follow the "Always have an OSS license", even less of an issue

**Review by mrjones-plip — CHANGES_REQUESTED** [review]
Love the bold move here with this PR!  

I think it's not ready as is because it will stifle key innovation by raising the barrier to entry too high.  See comments for details.

File: content/en/contribute/code/repository-checklist.md
Comment (andrablaj) [reply content/en/contribute/code/repository-checklist.md:31]: CD can be set up to deploy the changes to a development environment for testing before merging the branch to `main` and deploying the changes to production. I agree that we are not there yet, so I will remove this item from the list.

File: content/en/contribute/code/repository-checklist.md
Comment (andrablaj) [reply content/en/contribute/code/repository-checklist.md:44]: I assume "Anyone at Medic", right? Can people outside of Medic create repositories under Medic organization?

**Comment by andrablaj** [comment]
@garethbowen & @mrjones-plip, thank you for your comments, I am so happy we started this conversation! I failed to explain the thinking behind this checklist, so allow me to elaborate and sorry for not having done that better when opening this draft PR.

The checklist aims to identify items that all project repositories under the Medic organization should contain for quality and security. However, the purpose is not to add extra hassle et process but to agree on the items and create a [GitHub template repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) from those items. That way, every time a colleague creates a repo for a new project, they can use the template repository and have all the basic configurations out of the box without having to manually setup up branch protection rules, PR templates, licenses, basic CI build, etc. That would be time-saving rather than adding extra hassle and will create consistency and quality across Medic project repos. 

> This also means we have a grace period - there's a lot of effort in the items you've listed and it shouldn't all have to be done before you can create a repo, but it should all be done before we can call something a Product.

> it may be weeks or months until we met all the requirements on that list.

I agree that we shouldn't add an extra burden on the developers. However, the template repository could save time rather than give extra work. Once we agree on the items the repos should follow, we can create a `cht-template` repository with those configurations that can be used and customized (e.g. issue template content) according to the project's needs.
 
I do agree that the items on the list apply to the Medic products rather than all kinds of repos in general. That was my thinking when writing this (I was thinking `cht-interoperability`, `cht-sync`, `cht-monitoring`, etc.); thanks for pointing it out. 

Per your great suggestions, I separated the _must-haves_ - configurable via the `cht-template` repository, from the _Things to have to consider this Product has the quality we strive for_. I am happy to create that repository after we finalize the discussion on the _must-haves_.

How about this question?

> When to make a repo public vs private vs directory in existing (monolithic)?

**Review by mrjones-plip — APPROVED** [review]
Sweet! Thanks a bunch for incorporating the feedback. :shipit:

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:20]: This applies to the repo in general not just the main branch

```suggestion
- [ ] The following files:
```

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:21]:
```suggestion
    - `LICENSE` specifying AGPL-3.0 ([example](https://github.com/medic/cht-core/blob/master/LICENSE))
```

File: content/en/contribute/code/repository-checklist.md
Comment (garethbowen) [inline content/en/contribute/code/repository-checklist.md:58]: I think we should be more opinionated about open source.

```suggestion
A: Repositories should be public unless there is very good reason to make it private (e.g. the repository contains partner details that cannot be disclosed to public). Always keep in mind that it is much easier to start public than change to public later.
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Three small suggestions inline but love the direction this is going!

**Review by garethbowen — APPROVED** [review]
Awesome, thanks!

**Comment by andrablaj** [comment]
@garethbowen & @mrjones-plip Thank you for supporting with this issue, I am very happy with the checklist we came up together. 

I created a `cht-template` repository and started to add default configurations [here](https://github.com/medic/cht-template/pull/2). I will merge this PR once we finalize the configurations on the template repo.

**Pull request merged**
